### DISCUSSION TITLE
How to fetch DateTime null values

### DISCUSSION THREAD

**Discussion by TheVoid-0** [prisma/prisma#11256] [Q&A] [answered]
Hey there, thanks for the amazing work on Prisma, Me and my team are really excited to start using Prisma, but so far we've found a lot of difficulties adapting it for our needs (hard to make multiple databases with different schemas and we are currently having problems to resolve the schema path on the generated client), but anyway, I need some help with this error:  `Value out of range for the type. The column 'createdAt' contained an invalid datetime value with either day or month set to zero.`

This is hapenning with entries that have NULL values in `createdAt `column, however, in prisma schema i've set `createdAt `with `DateTime?`
```prisma
model users {
// other fields
createdAt                DateTime?                       @default(now()) @db.DateTime(0)
}
```

We are currently trying to migrate our services to use prisma, but null values in DateTimes are very common in our database, can someone elucidate me on why this happens and how I can workaround this issue? OBS: this happens when using: 
```ts
prismaClient.users.findFirst( { where: { id: someIdThatHasANullCreatedAt } } )
```

Thanks in advance

**Comment by austincrim** [comment, marked as answer]
Hey @TheVoid-0 👋 ,

~~This looks like an existing issue with Prisma. You can find more discussion and possible workarounds in #5006. Please add your +1 and specific use case in that thread if you feel it hasn't been covered. That will help our product and engineering teams prioritize in the future.~~

See reply 👇

**Reply by austincrim** [reply]
@TheVoid-0 After consulting with engineering, this looks to be a different issue than the one I linked above. If you could create a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml), our engineers will take it from there 👍.

Sorry for any confusion!

**Comment by TheVoid-0** [comment]
Thanks for answering @austincrim! I was just about to answer your first post when the second appeared! Alright I will try to gather some more info to create a bug report, maybe this weekend I will find some time to create a repository where this behavior can be reproduced

**Reply by TheVoid-0** [reply]
Sorry for the delay! I was not able to reproduce on a isolated database, so me and my team gave up for a while, now, with more time we came back at it and realized it was actually the first probem @austincrim mentioned! (#5006 ) Even though the SGBD and the application says the columns are NULL they are actually not... they have a 0000-00-00 date in it. So it's the database fault, we are trying our best right now to use prisma as isolated as possible, because it looks like this issue is all over the database unfortunately. Not sure if we will be able to fix the whole DB, but I guess it worth I try, we really want to use Prisma!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
